### PR TITLE
Fix process args fields

### LINKED ISSUE #1873: SedTransformer exec plugin script fails with space
I was following the [plugin docs](https://github.com/kubernetes-sigs/kustomize/tree/master/docs/plugins#examples) and creating a working version of the sed transformer exec plugin. With a simple sed expressions it was working fine. But with any expression containing space, the plugin was failing. 

The go glue code which extracts the plugin config and executes the plugin is correct, the issue is caused by incorrect argument handling in the [SedTransformer script](https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/someteam.example.com/v1/sedtransformer/SedTransformer)

Self-note: `argsFromFile` needs some documentation.

## Example without space - [SUCCESS]

sedTransformer.yaml:
```
apiVersion: someteam.example.com/v1
kind: SedTransformer
metadata:
  name: whatever
argsOneLiner: s/foo/bar/
```

## Example with space - [FAIL]

sedTransformer.yaml: 
```
apiVersion: someteam.example.com/v1
kind: SedTransformer
metadata:
  name: whatever
argsOneLiner: s/foo/bar baz/
```
#### Executed command
```
sed -e s/foo/bar -e baz/
```

## Example with argsFromFile - [FAIL]

Unfortunately putting sed args to file doesn't help:

sedTransformer.yaml: 
```
apiVersion: someteam.example.com/v1
kind: SedTransformer
metadata:
  name: whatever
argsFromFile: foo.sed
```

foo.sed:
```
s/foo/bar baz/
```

#### Executed command
```
sed -e s/foo/bar -e baz/
```

### PR DESCRIPTION
**Pull request by lalyos**
fixes #1873 

The original `strings.Split()` will fail for this:
```
apiVersion: someteam.example.com/v1
kind: SedTransformer
metadata:
  name: whatever
argsOneLiner: one two 'foo bar'
```

[coshell.Split()](https://godoc.org/github.com/gdm85/coshell/cosh#Split) handles it correctly

### CODE REVIEW

File: api/internal/plugins/execplugin/execplugin.go [hunk 1]
```diff
@@ -86,7 +87,7 @@ func (p *ExecPlugin) processOptionalArgsFields() error {
 	var c argsConfig
 	yaml.Unmarshal(p.cfg, &c)
 	if c.ArgsOneLiner != "" {
-		p.args = strings.Split(c.ArgsOneLiner, " ")
+		p.args, _ = cosh.Split(c.ArgsOneLiner)
```
[the remark this answers is not in the document]
Comment (lalyos) [reply api/internal/plugins/execplugin/execplugin.go]: fixed

**Comment by pwittrock** [comment]
This is cool.

/lgtm
/approve

**Comment by pwittrock** [comment]
FYI: Seeing this failure

# sigs.k8s.io/kustomize/api/internal/plugins/execplugin_test [sigs.k8s.io/kustomize/api/internal/plugins/execplugin.test]
internal/plugins/execplugin/execplugin_test.go:50:5: ldr.AddFile undefined (type ifc.Loader has no field or method AddFile)
ok  	sigs.k8s.io/kustomize/api/internal/plugins/compiler	16.517s
FAIL	sigs.k8s.io/kustomize/api/internal/plugins/execplugin [build failed]

File: api/internal/plugins/execplugin/execplugin_test.go [hunk 2]
```diff
@@ -43,9 +43,16 @@ s/$BAR/bar/g
 			"metadata": map[string]interface{}{
 				"name": "some-random-name",
 			},
-			"argsOneLiner": "one two",
+			"argsOneLiner": "one two 'foo bar'",
 			"argsFromFile": "sed-input.txt",
 		})
+
+	ldr.AddFile("/app/sed-input.txt", []byte(`
```
Comment (pwittrock) [inline api/internal/plugins/execplugin/execplugin_test.go]: @monopole did this function get removed in a recent factoring?

**Pull request merged**